### DISCUSSION TITLE
Count all steps in a pipeline

### DISCUSSION THREAD

**Discussion by havebeenfitz** [fastlane/fastlane#18563] [Q&A] [answered]
Hi! Is it possible to get a full count of tasks in a perfect scenario? Like here we have 15

![image](https://user-images.githubusercontent.com/31866271/114976563-d883e800-9e8e-11eb-8b32-7a7cc725fbcb.png)

We need this to calculate build progress and upload this progress to slack app. For now we're just hardcoding it, but it's a bit ugly.
Any help is appreciated.

**Comment by hjanuschka** [comment, marked as answer]
try 

```rb
Fastlane::Actions.executed_actions.length
```


its how the table is printed - see: https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/lane_manager_base.rb#L18

**Reply by havebeenfitz** [reply]
Thanks a lot! It must be it
